Shared/Mirror/mirrorClient.js

```diff
@@ -0,0 +1,314 @@
+		{ x: -0.5, y: 0.5 },	// Upper Left
+		{ x: 0.5, y: 0.5 },		// Upper Right
+		{ x: -0.5, y: -0.5 },	// Lower Left
+		{ x: 0.5, y: -0.5 },	// Lower Right
+		{ x: 0, y: 0 }			// Center
+	];
+	var lastHalfDimSignUsed = halfDimSigns[4];	// Contains one of the five objects in halfDimSigns that was last used when updating the overlay
+
+	// LOCAL FUNCTIONS
+	
+	// Takes in 'parsedMessage' which will have a property 'mirrorEditorsVisible'. If 'mirrorEditorsVisible' is true, then the mirror toggle overlay
+	// must be adjusted for the new dimensions/rotation of the mirror. 'parsedMessage' will have properties 'newPos', 'newRot', and 'newDimLength' to
+	// adjust the mirror toggle overlay. The mirror toggle overlay will also become visible If 'mirrorEditorsVisible' is false, then set the mirror toggle 
+	// overlay to invisible.
+	var adjustMirrorToggleOverlay = function(parsedMessage) {
+		if(parsedMessage.mirrorEditorsVisible) {
+			Overlays.editOverlay(mirrorToggleOverlayID, { position: parsedMessage.newPos });
+			Overlays.editOverlay(mirrorToggleOverlayID, { rotation: Quat.multiply(parsedMessage.newRot, { w: 0, x: -0.707, y: 0, z: -0.707}) });
+			Overlays.editOverlay(mirrorToggleOverlayID, { dimensions: { x: parsedMessage.newDimLength / 3, y: parsedMessage.newDimLength, z: parsedMessage.newDimLength } });
+			Overlays.editOverlay(mirrorToggleOverlayID, { visible: true });			
+		} else {
+			Overlays.editOverlay(mirrorToggleOverlayID, { visible: false });
+		} 	
+	}
+	
+	// Checks the distance between the user and the mirror and reveal the mirror editors when the user is within the 'avatarDistanceThreshold' range
+	var checkAvatarDistance = function() {
+		var mirrorProps = Entities.getEntityProperties(_this.entityID, ["position"]);
+		var avatarDistance = Math.abs(Vec3.distance(MyAvatar.position, mirrorProps.position));
+		if(avatarDistance < avatarDistanceThreshold) {
+			Messages.sendMessage(clientToServerChannel, JSON.stringify({ serverFunction: "updateHeartbeat", withinThreshold: true }));
+		} else if (avatarDistance > avatarDistanceThreshold) {
+			Messages.sendMessage(clientToServerChannel, JSON.stringify({ serverFunction: "updateHeartbeat", withinThreshold: false }));
+		}
+	}
+	
+	// Takes in the spectator camera position and creates an array of the front four vertices of the mirror. It then calculates
+	// the distance between each vertex and the spectator camera position and returns the farthest distance.
+	function findNearClipPlaneDistance(spectatorCameraPos) {		
+		var mirrorProps = Entities.getEntityProperties(_this.entityID, ["position", "dimensions", "rotation"]);
+		var mirrorFrontVertices = [];	// An array containing the positions of the front 4 vertices on the mirror, where 0 is the
+										// Upper Right, 1 is the Lower Right, 2 is the Upper Left, and 3 is Lower Left
+		var vertexPosX;
+		var vertexPosY;
+		var vertexPosZ;
+
+		for(var i = 0; i < 2; i++) {
```

Comment:
2 is a magic number, please make a constant / variable that defines what it means.